luhn 1.0.0.2: only test isValid, use (most) x-common cases

petertseng opened the pull request:
https://github.com/exercism/x-common/pull/474
https://github.com/exercism/x-common/pull/496
https://github.com/exercism/x-common/pull/522
https://github.com/exercism/x-common/pull/547

In consideration of https://github.com/exercism/x-common/issues/523, we
will not be testing the invalid inputs. They have been commented out.

Closes #469

petertseng commented:
Say the word if you disagree with the path taken. I don't want to have to do extra work to filter out the invalid inputs only to have to delete the work if https://github.com/exercism/x-common/issues/523 is accepted.

petertseng commented:
and sorry I hadn't set up hlint on this machine

petertseng commented on exercises/luhn/test/Tests.hs:
```diff
@@ -1,79 +1,73 @@
-
-      it "create 232320057766355" $
-        create 232320057766355 `shouldBe` 2323200577663554
+specs = describe "valid" $ for_ cases test
```
notice! no luhn!

rbasso commented on exercises/luhn/test/Tests.hs:
```diff
@@ -1,79 +1,73 @@
+--        , Case { description = "valid strings with punctuation included become invalid"
+--               , input       = "055-444-285"
+--               , expected    = False
+--               }
```
Here it seems that you suppressed one test:

```json
    {
      "description": "valid strings with symbols included become invalid",
      "property": "valid",
      "input": "055£ 444$ 285",
      "expected": false
    },
```

Was it intentional?

I don't consider that a problem, because considering what was discussed in https://github.com/exercism/x-common/issues/428, it probably shouldn't exist anyway.

petertseng replied:
Oh! I guess I'll add it (and comment it) for the purpose of following x-common, but you are right that we expect the case to go away soon.

petertseng commented:
Oh interesting. @rbasso could I trouble you to approve this one again? Since I made a change since the previous approval, and this matters if I'm octomerging. Interestingly, the fact that I made changes since the previous approval wouldn't stop me from merging on the web, which actually makes this seem a little inconsistent.
